tests/Unit/DataStructures/Tensor/Expressions/Test_Product.cpp

```diff
@@ -1234,6 +1234,100 @@ void test_spatial_spacetime_index(const DataType& used_for_size) noexcept {
   }
 }
 
+// \brief Test the products of tensors where concrete time indices are used for
+// spacetime indices
+//
+// \details
+// The product cases tested are:
+// - \f$L = R_{t} * S * R_{t}\f$
+// - \f$L^{b} = G_{t}{}^{a} * H_{a}{}^{tb}\f$
+// - \f$L^{T}{}_{ba} = R_{a} * R_{b}\f$
+// - \f$L_^{ct} = G_{t}{}^{b} * R_{b} * H_{t}^{ta}\f$
+//
+// \tparam DataType the type of data being stored in the tensor operand of the
+// products
+template <typename DataType>
+void test_time_index(const DataType& used_for_size) noexcept {
+  Tensor<DataType, Symmetry<1>,
+         index_list<SpacetimeIndex<3, UpLo::Lo, Frame::Grid>>>
+      R(used_for_size);
+  assign_unique_values_to_tensor(make_not_null(&R));
+
+  Tensor<DataType> S{{{used_for_size}}};
+  if constexpr (std::is_same_v<DataType, double>) {
+    // Replace tensor's value from `used_for_size` to a proper test value
+    S.get() = -2.2;
+  } else {
+    assign_unique_values_to_tensor(make_not_null(&S));
+  }
+
+  // \f$L = R_{t} * S * R_{t}\f$
+  const Tensor<DataType> L =
+      TensorExpressions::evaluate(R(ti_t) * S() * R(ti_t));
+  CHECK(L.get() == R.get(0) * S.get() * R.get(0));
+
+  Tensor<DataType, Symmetry<2, 1>,
+         index_list<SpacetimeIndex<3, UpLo::Lo, Frame::Grid>,
+                    SpacetimeIndex<3, UpLo::Up, Frame::Grid>>>
+      G(used_for_size);
+  assign_unique_values_to_tensor(make_not_null(&G));
+
+  Tensor<DataType, Symmetry<3, 2, 1>,
+         index_list<SpacetimeIndex<3, UpLo::Lo, Frame::Grid>,
+                    SpacetimeIndex<3, UpLo::Up, Frame::Grid>,
+                    SpacetimeIndex<3, UpLo::Up, Frame::Grid>>>
+      H(used_for_size);
+  assign_unique_values_to_tensor(make_not_null(&H));
+
+  // \f$L^{b} = G_{t}{}^{a} * H_{a}{}^{tb}\f$
+  const Tensor<DataType, Symmetry<1>,
+               index_list<SpacetimeIndex<3, UpLo::Up, Frame::Grid>>>
+      L_B = TensorExpressions::evaluate<ti_B>(G(ti_t, ti_A) *
+                                              H(ti_a, ti_T, ti_B));
+
+  for (size_t b = 0; b < 4; b++) {
+    DataType expected_product = make_with_value<DataType>(used_for_size, 0.0);
+    for (size_t a = 0; a < 4; a++) {
+      expected_product += G.get(0, a) * H.get(a, 0, b);
+    }
+    CHECK_ITERABLE_APPROX(L_B.get(b), expected_product);
+  }
+
+  Tensor<DataType, Symmetry<2, 1, 1>,
+         index_list<SpacetimeIndex<3, UpLo::Up, Frame::Grid>,
+                    SpacetimeIndex<3, UpLo::Lo, Frame::Grid>,
+                    SpacetimeIndex<3, UpLo::Lo, Frame::Grid>>>
+      L_Tba(used_for_size);
+  // \f$L^{T}{}_{ba} = R_{a} * R_{b}\f$
+  TensorExpressions::evaluate<ti_T, ti_b, ti_a>(make_not_null(&L_Tba),
+                                                R(ti_a) * R(ti_b));
+
+  for (size_t b = 0; b < 4; b++) {
+    for (size_t a = 0; a < 4; a++) {
+      CHECK_ITERABLE_APPROX(L_Tba.get(0, b, a), R.get(a) * R.get(b));
```

Comment:
shouldn't you also check that the other components are unchanged? This is probably true for other test files and tests